src/extensions/default/QuickView/main.js

```diff
@@ -784,10 +812,15 @@ define(function (require, exports, module) {
 
     // Setup initial UI state
     setEnabled(prefs.get("enabled"), true);
+    setExtensionlessImagePreview(prefs.get("extensionlessImagePreview"));
```

Comment:
So the whole `doNotSave` dance was done for the initial load case?  I'll echo that for this as well and avoid the write.